server/api.go

```diff
@@ -350,7 +350,7 @@ func (a *API) oauthRedirectHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if mmUser.Id != a.p.GetBotUserID() && msteamsUser.Mail != mmUser.Email {
```

Comment:
This change shouldn't be needed because the client layer should be taking care of that entirely.